test/e2e/collection/namespace_filtering_test.go

```diff
@@ -0,0 +1,182 @@
+					},
+				},
+				Pipelines: []logging.PipelineSpec{
+					{
+						Name:       "test-app",
+						OutputRefs: []string{elasticsearch.Name},
+						InputRefs:  []string{"application-logs"},
+					},
+				},
+			},
+		}
+		if err := e2e.CreateClusterLogForwarder(forwarder); err != nil {
+			Fail(fmt.Sprintf("Unable to create an instance of clusterlogforwarder: %v", err))
+		}
+
+	})
+
+	GetNamespaces := func() ([]string, error) {
+		options := metav1.ListOptions{
+			LabelSelector: "component=elasticsearch",
+		}
+		app_index := "app-000001"
+		pods, err := e2e.KubeClient.CoreV1().Pods(constants.OpenshiftNS).List(context.TODO(), options)
+		if err != nil {
+			return nil, errors.New("Getting pods error")
+		}
+		if len(pods.Items) == 0 {
+			return nil, errors.New("No pods found for elasticsearch")
+		}
+		log.V(3).Info("Pod ", "PodName", pods.Items[0].Name)
+		query := "--query=" + app_index + "/_search?size=10000"
+		stdout, err := e2e.PodExec(constants.OpenshiftNS, pods.Items[0].Name, "elasticsearch", []string{"es_util", query})
+		if err != nil {
+			return nil, errors.New("Pods exec error")
+		}
+		// Get total hits
+		var jsonMap map[string]interface{}
+		if err := json.Unmarshal([]byte(stdout), &jsonMap); err != nil {
+			return nil, errors.New("json unmarshall error")
+		}
+		total := jsonMap["hits"].(map[string]interface{})["total"]
+		val := int(total.(float64))
+
+		namespaceMap := make(map[string]bool)
+		var namespaceList []string
+		// Parse each document and extract namespace value
+		if val > 0 {
+			docs := jsonMap["hits"].(map[string]interface{})["hits"].([]interface{})
+			for _, document := range docs {
+				namespace_value := document.(map[string]interface{})["_source"].(map[string]interface{})["kubernetes"].(map[string]interface{})["namespace_name"].(string)
+
+				if _, found := namespaceMap[namespace_value]; !found {
+					namespaceMap[namespace_value] = true
+					namespaceList = append(namespaceList, namespace_value)
+				}
+			}
+		}
+		return namespaceList, nil
+	}
+
+	AssertBehaviours := func() {
+		Expect(e2e.LogStores[elasticsearch.GetName()].HasApplicationLogs(framework.DefaultWaitForLogsTimeout)).To(BeTrue(), "Expected to find stored application logs")
+
+		namespaceList, err := GetNamespaces()
+		Expect(err).To(BeNil(), fmt.Sprintf("Error fetching logs: %v", err))
+		Expect(namespaceList).NotTo(BeNil())
+		Expect(len(namespaceList)).To(Equal(1))
+		Expect(namespaceList[0]).To(Equal(appNamespace1))
+	}
+
+	DescribeTable("Running collector tests",
+		func(collectorType helpers.LogComponentType) {
+			cr := helpers.NewClusterLogging(collectorType)
+			if err := e2e.CreateClusterLogging(cr); err != nil {
+				Fail(fmt.Sprintf("Unable to create an instance of cluster logging: %v", err))
+			}
+			if err := e2e.WaitFor(collectorType); err != nil {
+				Fail(fmt.Sprintf("Failed waiting for component %s to be ready: %v", collectorType, err))
+			}
+			AssertBehaviours()
```

Comment:
Consider replacing this with the function content;  I don't see the advantage to move it into a separate function